### ISSUE TITLE
Email recieved from Google titled "Critical security alert"

### ISSUE THREAD

**Issue by Obegg** [yt-dlp/yt-dlp#8267]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2023.09.24** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

I know it will sound far fetched but hear me out.
I've been using mpv + yt-dlp for quite a while now, and I had no issues.
Clean Windows 10 install and no viruses, very minimal setup.
Then I discovered the `--cookies-from-browser` (Firefox), which I used and it's working fine,
After 3-6 weeks of daily driving the cookies settings I received a email from Google,
Title: `Critical security alert`
Message: `Suspicious activity in your account`
`Someone might have accessed your Google Account using harmful malware on one of your devices. You’ve been signed out on that device for your protection.`
And I find out my session on my browser is logged out, so either way I change password and do a fresh Windows 10 install, this time not using the cookies feature, works fine for a month, then I use the cookies feature and again after 3-6 weeks same email and same message... Then I tested it for the third time, another fresh install of Windows 10, enable cookies feature on yt-dlp and again same email.
So... I tried to search on yt-dlp GitHub if there is a known issue or an opened open and found no results,
It seems like yt-dlp is doing something with cookies that triggers Google alerts,
I do wonder why it seems like it's only happening to me (since there are no issues on GitHub related to this).

### Provide verbose output that clearly demonstrates the problem

- [ ] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

**Comment by gamer191** [comment]
That is quite interesting 🤔

May I ask why you did a fresh install? Did it block you from logging back in?

**Comment by bashonly** [comment]
We don't know that the security alerts are related to yt-dlp usage. It could be any number of things (e.g. are you signed into the account on any other devices?), and google isn't exactly transparent about what is flagging the alerts.

Try creating a google/youtube account only for use with yt-dlp, and *only* use yt-dlp with that account for a while. If you receive security alerts for that account, then we'll know the alerts are definitely related to yt-dlp usage.

**Comment by gamer191** [comment]
> google isn't exactly transparent about what is flagging the alerts

I wonder if you could request further information about it using https://takeout.google.com/. Two data points that could be relevant are "access log activity" and "google account"

**Comment by bashonly** [comment]
FYI, `--mark-watched` is currently broken for some (all?) users: #8236

Maybe try removing it from your config/commands and seeing if it was the trigger

**Comment by absidue** [comment]
@Obegg Well the main question is whether the suspicious activity is just because you are using your account with yt-dlp or specifically because of the mark as watched flag. The only way to really narrow that down, would be to keep using the cookies from browser flag, without the mark as watched one and see if you still get flagged. If you don't get flagged without the mark as watched feature, the problem is with the mark as watched feature.

The logs you mentioned about the many retries would probably also be useful to share, because that definitely sounds like something that could trip YouTube's suspicious activity monitoring.

**Comment by QuickHare** [comment]
When using cookies from another source (like a browser), you are using the same login credentials. Google is using analysis to determine if there's any foul play by comparing it to other activity on the same account.

As yt-dlp, as well as youtube-dl and other programs that download from YouTube, doesn't look the same as your browser session, Google can get a little itchy about it. Google also has something against yt-dlp and youtube-dl entirely, so goes out their way to detect and close down usage of them.

`--mark-watched` does have a bug associated with it that might be causing too much communication that Google thinks looks suspicious.

I've tried to make mine look as close to the usual browser as possible, including setting a user agent where I can, so it makes websites with accounts less suspicious of me using a different program with the same credentials.

**Comment by bashonly** [comment]
That's pretty normal ffmpeg HLS junk output, nothing to be concerned about

**Comment by bashonly** [comment]
> You say you think `--mark-watched` is the culprit

When you pass this option, yt-dlp sends a request one of youtube's *tracking servers* and effectively increments that video's view count. Surely you can see why youtube might care more about automated usage of this moreso than, say, HLS segments being requested from their CDN

**Comment by gamer191** [comment]
I'm not sure how relevant it is, but are you using a managed enterprise account (an account that doesn't end in @gmail.com)?

**Comment by gamer191** [comment]
> Or maybe because my browser extensions (I have no idea how that's related), or maybe because I use Pi-Hole?

@Obegg can you please try just not using `--cookies-from-browser` at all, and see whether you get that email? If you've already used yt-dlp since logging back in, can you please log out of your browser and then log back in (if you want, feel free to **also** clear your cookies, but make sure to log out using youtube's logout button first)

**Comment by gamer191** [comment]
@Obegg Actually, I have a better idea:
1. log out and log back in, just for good measure
2. export your cookies, then use them with yt-dlp, using --cookies
3. clear your youtube/google cookies from firefox (without first logging out)
4. log in again
5. wait and see whether firefox or yt-dlp gets logged out

**Comment by gamer191** [comment]
Can you try again, and this time also clear your cookies for all google.com websites?

**Comment by gamer191** [comment]
Actually, I have a better idea:
1. open about:addons
2. make sure all your addons (except cookies.txt) are disabled whilst in incognito mode (I don't care if you trust them)
3. open an incognito window
4. log into youtube.com and export your cookies
5. change your mpv config to use those cookies
6. wait

**Comment by gamer191** [comment]
`ALL`, I guess

**Comment by amishmm** [comment]
This happened with me today!

First I got scared that someone hacked in to my account or laptop.

But later when yt-dlp stopped working, I realized that Google logged that session out.

I am using yt-dlp with cookies and that session got logged out. (Shown as windows machine by Google)

To solve it again, I did following:

1. Create new container with name **yt-dlp** in Firefox
2. Login to Youtube (in **yt-dlp** container) and then close Firefox
3. Run: `yt-dlp --cookies-from-browser firefox::yt-dlp --cookies ytdl-cookies.txt`
4. Use the generated cookie file when running yt-dlp (via --cookies option)
5. Open firefox again and delete yt-dlp container, to avoid it conflicting with yt-dlp cookies.
